### DISCUSSION TITLE
How to get HTML and CSS of a specific Component

### DISCUSSION THREAD

**Discussion by codingmachine16** [GrapesJS/grapesjs#3669] [General]
Hi @artf 
I searched a component from editor through below code 
      const wrapper = editor.getWrapper();
      const el = wrapper.find("#header-placeholder")[0];
      console.log(el);
Now I want to get HTML and CSS of this component like 
      console.log(el.getHtml());  and       console.log(el.getCss());

How is is possible? I want to get html and css of specific component and store in DB table

but el.getHtml() and el.getCss() are generating error. Please suggest.

**Comment by codingmachine16** [comment]
The solution is

const wrapper = editor.getWrapper();
const el = wrapper.find("#header-placeholder")[0];
const compHtml=el.toHTML();

**Comment by mosh-tudor** [comment]
For styles you can use:

`editor.CodeManager.getCode(yourComponent, 'css', { cssc: editor.CssComposer })`
